staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -47,7 +47,7 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 			req = req.WithContext(authenticator.WithAudiences(req.Context(), apiAuds))
 		}
 		resp, ok, err := auth.AuthenticateRequest(req)
-		defer recordAuthMetrics(req.Context(), resp, ok, err, apiAuds, authenticationStart)
+		recordAuthMetrics(req.Context(), resp, ok, err, apiAuds, authenticationStart)
```

Comment:
/hold

This breaks recording of the authentication audience error case on line 60. The fact that unit tests are green mean we have a test gap that should be closed as well.

Once this is correct, I would support merge and backport

Reply:
I've added tests. Also, this error on line 60 was not working before. When you do `defer f(err)` err is evaluated at the time of defer, not later.